src/libraries/Common/src/Interop/Android/System.Security.Cryptography.Native.Android/Interop.Cipher.cs

```diff
@@ -0,0 +1,222 @@
+            {
+                throw CreateOpenSslCryptographicException();
+            }
+        }
+
+        [DllImport(Libraries.CryptoNative, EntryPoint = "AndroidCryptoNative_CipherDestroy")]
+        internal static extern void EvpCipherDestroy(IntPtr ctx);
+
+        [DllImport(Libraries.CryptoNative, EntryPoint = "AndroidCryptoNative_CipherReset")]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        internal static extern bool EvpCipherReset(SafeEvpCipherCtxHandle ctx);
+
+        [DllImport(Libraries.CryptoNative, EntryPoint = "AndroidCryptoNative_CipherCtxSetPadding")]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        internal static extern bool EvpCipherCtxSetPadding(SafeEvpCipherCtxHandle x, int padding);
+
+        [DllImport(Libraries.CryptoNative, EntryPoint = "AndroidCryptoNative_CipherUpdate")]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        private static extern bool EvpCipherUpdate(
+            SafeEvpCipherCtxHandle ctx,
+            ref byte @out,
+            out int outl,
+            ref byte @in,
+            int inl);
+
+        internal static bool EvpCipherUpdate(
+            SafeEvpCipherCtxHandle ctx,
+            Span<byte> output,
+            out int bytesWritten,
+            ReadOnlySpan<byte> input)
+        {
+            return EvpCipherUpdate(
+                ctx,
+                ref MemoryMarshal.GetReference(output),
+                out bytesWritten,
+                ref MemoryMarshal.GetReference(input),
+                input.Length);
+        }
+
+        [DllImport(Libraries.CryptoNative, EntryPoint = "AndroidCryptoNative_CipherUpdateAAD")]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        private static extern bool EvpCipherUpdateAAD(
+            SafeEvpCipherCtxHandle ctx,
+            ref byte @in,
+            int inl);
+
+        internal static bool EvpCipherUpdateAAD(
+            SafeEvpCipherCtxHandle ctx,
+            ReadOnlySpan<byte> input)
+        {
+            return EvpCipherUpdateAAD(
+                ctx,
+                ref MemoryMarshal.GetReference(input),
+                input.Length);
+        }
+
+        [DllImport(Libraries.CryptoNative, EntryPoint = "AndroidCryptoNative_CipherFinalEx")]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        private static extern bool EvpCipherFinalEx(
+            SafeEvpCipherCtxHandle ctx,
+            ref byte outm,
+            out int outl);
+
+        internal static bool EvpCipherFinalEx(
+            SafeEvpCipherCtxHandle ctx,
+            Span<byte> output,
+            out int bytesWritten)
+        {
+            return EvpCipherFinalEx(ctx, ref MemoryMarshal.GetReference(output), out bytesWritten);
+        }
+
+        [DllImport(Libraries.CryptoNative, EntryPoint = "AndroidCryptoNative_CipherSetTagLength")]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        internal static extern bool EvpCipherSetTagLength(
```

Comment:
I'll give a vague encouragement here for avoiding Evp on any members that aren't called from both Unix and Android code.  Since this is a function that doesn't exist in the Unix/OpenSSL one, it doesn't need the prefix.